src/main/java/com/ericsson/ei/handlers/EventToObjectMapHandler.java

```diff
@@ -83,28 +83,29 @@ public void setJmesPathInterface(JmesPathInterface jmesPathInterface) {
 
     public void updateEventToObjectMapInMemoryDB(RulesObject rulesObject, String event, String objectId) {
         String eventId = getEventId(rulesObject, event);
-        String condition = "{\"_id\" : \"" + eventId + "\"}";
-        ArrayList<String> list =  getEventToObjectList(eventId);
-        boolean firstTime = list.isEmpty();
-        list = updateList(list, eventId, objectId);
-        ObjectMapper mapper = new ObjectMapper();
-        JsonNode entry = null;
-
+        String condition = "{\"_id\" : \"" + objectId + "\"}";
+        LOGGER.debug("Checking document exists in the collection with ID : {}", condition );
+        LOGGER.debug("EventID : {}", eventId);
+        boolean docExists = mongodbhandler.checkObjectExists(databaseName, collectionName, condition);
         try {
-            entry = new ObjectMapper().readValue(condition, JsonNode.class);
-            ArrayNode jsonNode = mapper.convertValue(list, ArrayNode.class);
-            ((ObjectNode) entry).set(listPropertyName, mapper.readTree(jsonNode.toString()));
-            String mapStr = entry.toString();
-            LOGGER.debug("MongoDbHandler Insert/Update Event: {}\nto database: {} and to Collection: {}", mapStr, databaseName, collectionName);
-            if (firstTime) {
-                mongodbhandler.insertDocument(databaseName, collectionName, mapStr);
+        	if (!docExists) {
+        		ArrayList<String> list =  new ArrayList<String>();
+        		list.add(eventId);
+        		final ObjectMapper mapper = new ObjectMapper();
+    			JsonNode entry = new ObjectMapper().readValue(condition, JsonNode.class);
+        		ArrayNode jsonNode = mapper.convertValue(list, ArrayNode.class);
+        		((ObjectNode) entry).set(listPropertyName, mapper.readTree(jsonNode.toString()));
+                final String mapStr = entry.toString(); 
+            	LOGGER.debug("MongoDbHandler Insert/Update Event: {}\nto database: {} and to Collection: {}", mapStr, databaseName, collectionName);	
+            	mongodbhandler.insertDocument(databaseName, collectionName, mapStr );
             } else {
-                mongodbhandler.updateDocument(databaseName, collectionName, condition, mapStr);
+                mongodbhandler.updateDocumentAddToSet(databaseName, collectionName, condition, eventId);
             }
         } catch (Exception e) {
             LOGGER.info("Failed to update event object list.", e);
```

Comment:
Perhaps info is not the best log level when an error occurs.

Reply:
changed to error. Haven't modified the already existing code but have changed to error now.